### PR TITLE
print warning when developer is using older version

### PR DESCRIPTION
**Pull request by bsudekum**
Closes: https://github.com/mapbox/mapbox-navigation-ios/issues/1003

This prints a gentle reminder to the console when a developer using an older version of the sdk.
This will be a no-op until the documentation is updated. 

/cc @mapbox/navigation-ios

### CODE REVIEW

File: MapboxCoreNavigation/RouteController.swift [hunk 1]
```diff
@@ -720,6 +722,23 @@ extension RouteController: CLLocationManagerDelegate {
             strongSelf.delegate?.routeController?(strongSelf, didRerouteAlong: route)
         }
     }
+    
+    func checkForUpdates() {
+        #if TARGET_IPHONE_SIMULATOR
+            guard let latestVersion = String(describing: Bundle(for: RouteController.self).object(forInfoDictionaryKey: "CFBundleShortVersionString")) else { return }
+            URLSession.shared.dataTask(with: URL(string: "https://www.mapbox.com/mapbox-navigation-ios/navigation/latest_version")!, completionHandler: { (data, response, error) in
```
Comment (1ec5) [inline MapboxCoreNavigation/RouteController.swift]: We should put something at this URL before releasing with this PR.

File: scripts/document.sh [hunk 2]
```diff
@@ -50,4 +50,7 @@ REPLACE_REGEXP+='s/BRANDLESS_DOCSET_TITLE/Navigation SDK for iOS $1/, '
 REPLACE_REGEXP+="s/<span class=\"kt\">(${DIRECTIONS_SYMBOLS})<\/span>/<span class=\"kt\"><a href=\"${BASE_URL//\//\\/}\/directions\/${DIRECTIONS_VERSION}\/Classes\/\$1.html\">\$1<\/a><\/span>/, "
 
 find ${OUTPUT} -name *.html -exec \
-    perl -pi -e "$REPLACE_REGEXP" {} \;
\ No newline at end of file
+    perl -pi -e "$REPLACE_REGEXP" {} \;
+
+
+echo $SHORT_VERSION > ./latest_version
```
Comment (1ec5) [inline scripts/document.sh]: This puts the file in the current working directory, which isn’t the same as the output directory (`$OUTPUT`).

File: MapboxCoreNavigation/RouteController.swift [hunk 3]
```diff
@@ -720,6 +722,23 @@ extension RouteController: CLLocationManagerDelegate {
...
+                if let _ = error { return }
+                guard let httpResponse = response as? HTTPURLResponse, httpResponse.statusCode == 200 else { return }
+                
+                guard let data = data, let currentVersion = String(data: data, encoding: .utf8) else { return }
+                
+                if latestVersion != currentVersion {
```
Comment (1ec5) [inline MapboxCoreNavigation/RouteController.swift]: (Note that this string comparison won’t correctly handle any prereleases we do.)

Comment (bsudekum) [reply MapboxCoreNavigation/RouteController.swift]: Yeah I think thats fine.

File: MapboxCoreNavigation/RouteController.swift (hunk 1 above)
Comment (bsudekum) [reply MapboxCoreNavigation/RouteController.swift]: Added.

Comment (bsudekum) [reply MapboxCoreNavigation/RouteController.swift]: https://www.mapbox.com/mapbox-navigation-ios/navigation/latest_version

File: MapboxCoreNavigation/RouteController.swift [hunk 4]
```diff
@@ -720,6 +722,24 @@ extension RouteController: CLLocationManagerDelegate {
             strongSelf.delegate?.routeController?(strongSelf, didRerouteAlong: route)
         }
     }
+    
+    func checkForUpdates() {
+        #if TARGET_IPHONE_SIMULATOR
+        guard let version = Bundle(for: RouteController.self).object(forInfoDictionaryKey: "CFBundleShortVersionString") else { return }
+        let latestVersion = String(describing: version)
```
Comment (1ec5) [inline MapboxCoreNavigation/RouteController.swift]: Nit: inconsistent indentation.

File: scripts/document.sh [hunk 5]
```diff
@@ -53,3 +53,6 @@ REPLACE_REGEXP+="s/<span class=\"kt\">(${DIRECTIONS_SYMBOLS})<\/span>/<span clas
 
 find ${OUTPUT} -name *.html -exec \
     perl -pi -e "$REPLACE_REGEXP" {} \;
+
+
+echo $SHORT_VERSION > $output/latest_version
```
Comment (1ec5) [inline scripts/document.sh]: `$output` ≠ `$OUTPUT`

**Pull request merged**